unlock-js/src/__tests__/deploy.test.js

```diff
@@ -114,29 +40,46 @@ describe('contract deployer', () => {
     status: '0x1',
   }
   beforeEach(() => {
-    accountsAndYield(unlockAccountsOnNode)
-    ethGasPriceAndYield()
-    sendTransactionAndYield(unlockAccountsOnNode[0], transaction.hash)
-    ethGetTransactionReceiptAndYield(transaction.hash, transactionReceipt)
-    ethGetCodeAndYield('0xbbbdeed4c0b861cb36f4ce006a9c90ba2e43fdc2', 'latest')
-    ethGasPriceAndYield()
-    sendTransactionAndYield(
-      unlockAccountsOnNode[0],
-      transaction.hash,
-      '0xc4d66de8000000000000000000000000aaadeed4c0b861cb36f4ce006a9c90ba2e43fdc2', // data (no idea what this is, but there it is)
-      '0xbbbdeed4c0b861cb36f4ce006a9c90ba2e43fdc2', // contract address
-      '0xf4240' // gas
+    nock.accountsAndYield(unlockAccountsOnNode)
+    nock.ethGasPriceAndYield(gasPrice)
+    // contract deploy call
+    nock.ethSendTransactionAndYield(
+      {
+        from: unlockAccountsOnNode[0],
+        data: Unlock.bytecode,
+        gas: '0x' + GAS_AMOUNTS.deployContract.toString(16),
+      },
+      gasPrice,
+      transaction.hash
+    )
+    nock.ethGetTransactionReceipt(transaction.hash, transactionReceipt)
+    // get the contract bytecode
+    nock.ethGetCodeAndYield(
+      '0xbbbdeed4c0b861cb36f4ce006a9c90ba2e43fdc2',
+      'latest',
+      Unlock.bytecode
+    )
+    nock.ethGasPriceAndYield(gasPrice)
+    // initialize call
+    nock.ethSendTransactionAndYield(
+      {
+        to: '0xbbbdeed4c0b861cb36f4ce006a9c90ba2e43fdc2', // contract address
+        from: unlockAccountsOnNode[0],
+        data:
+          '0xc4d66de8000000000000000000000000aaadeed4c0b861cb36f4ce006a9c90ba2e43fdc2', // data (no idea what this is, but there it is)
```

Comment:
I think we need to clarify what this is otherwise we would have no idea why this may fail?

Reply:
I understand this is the behavior previously but if we're refactoring, we might as well clean things up and remove some tech debt.

Reply:
So I think we'd first need to understand which call in the deploy script makes this specific request. Then we can probably have a better way of understanding what it is and "rebuild" it from there?